src/Fixer/PhpBasic/Feature/VisibilityPropertiesFixer.php

```diff
@@ -258,30 +260,38 @@ private function isPropertyInAnotherClass(string $variableName, ?string $classNa
 
     private function getPropertyVariable(Tokens $tokens, int $key, bool $propertyExclusion): ?string
     {
-        $previousTokenIndex = $tokens->getPrevMeaningfulToken($key);
-        $previousPreviousTokenIndex = $tokens->getPrevMeaningfulToken($previousTokenIndex);
-        if (
-            $tokens[$previousTokenIndex]->isGivenKind([T_PUBLIC, T_PROTECTED, T_PRIVATE])
-            || (
-                $tokens[$previousTokenIndex]->isGivenKind(T_STATIC)
-                && $tokens[$previousPreviousTokenIndex]->isGivenKind([T_PUBLIC, T_PROTECTED, T_PRIVATE])
-            )
-        ) {
+        /** @var array<Token> $previousTokens */
+        $index = $key;
+
+        for ($i = 0; $i < 4; $i++) {
+            $previousTokenIndex = $tokens->getPrevMeaningfulToken($index);
+            $previousTokens[] = $tokens[$previousTokenIndex];
+            $index = $previousTokenIndex;
+        }
+
+        $visibilityToken = null;
+
+        foreach ($previousTokens as $previousToken) {
+            if ($previousToken->isGivenKind([T_PUBLIC, T_PROTECTED, T_PRIVATE, 10028, 10029, 10030])) {
+                $visibilityToken = $previousToken;
+                break;
+            }
+
+            if ($previousToken->isGivenKind(T_FUNCTION)) {
+                break;
+            }
+        }
```

Comment:
Private method

Reply:
Sorry, I do not quite understand your comment.

Reply:
This code should be moved to private method that returns `$visibilityToken`

Reply:
Now I got you and you are right. I was thinking about this too. It is fixed now.